### PR TITLE
Rename AnalyzeRequest to AnalyzeTextOptions

### LINKED ISSUE #12486: Resolve issues found during API unification review
From https://apiview.dev/Assemblies/Review/d40a51d26cf2444a85d302758ca3a2d1:

- [x] Rename `AnalyzeRequest` to `AnalyzeTextRequest` to fit method name.
- [x] Mention in documentation for that model that one of `Analyzer` or `Tokenizer` is required (but do not validate, per guidelines about avoid tight coupling between client and server behaviors - let the server validate).
- [x] Remove `SynonymMap.Formatter` property

### LINKED ISSUE #12530: Rename AnalyzeRequest to AnalyzeTextOptions, and add constructor overloads / helpers for required fields
Per discussion,

- [x] Rename `AnalyzeRequest` to `AnalyzeTextOptions`. `SearchIndexClient.AnalyzeText` should look like: `AnalyzeText(string indexName, AnalyzeTextOptions options, ...)`.
- [x] Add constructor overloads or other initialization helpers to require `text`, and either `analyer` or `tokenizer` (or, if you want, `analyzerName`, and `tokenizerName`, though I'll open up a separate issue for that).

```
class AnalyzeTextOptions {
  ctor(string text, LexicalAnalyzerName analyzer);
  ctor(string text, LexicalTokenizerName tokenizer);
  string text { get; }
  LexicalAnalyzerName? analyzer { get; }
  LexicalTokenizerName? tokenizer { get; }
  (remaining properties: { get; set; })
```

### LINKED ISSUE #12536: Rename properties and parameters for analyzer, tokenizer, and synonymMaps to *Name(s)
To be consistent with past renames for SearchField, SearchIndex, et. al, please make sure that all properties and parameters of type `LexicalAnalyzerName` or `LexicalTokenizerName`, and any string or string[] parameters for synonym map name(s) end in "Name" or "Names", as appropriate.

The following list was generated from .NET. Some renames have already been done, but please check in your code all of them:

Class | Type | Method | Name
--- | --- | --- | ---
AnalyzeTextOptions | Property |  | AnalyzerName
AnalyzeTextOptions | Property |  | Tokenizer
AnalyzeTextOptions | Parameter | .ctor | analyzerName
AnalyzeTextOptions | Parameter | .ctor | tokenizerName
CustomAnalyzer | Property |  | Tokenizer
CustomAnalyzer | Parameter | .ctor | tokenizer
SearchField | Property |  | AnalyzerName
SearchField | Property |  | SearchAnalyzerName
SearchField | Property |  | IndexAnalyzerName
SearchField | Property |  | SynonymMapNames
SearchableField | Property |  | AnalyzerName
SearchableField | Property |  | SearchAnalyzerName
SearchableField | Property |  | IndexAnalyzerName
SearchableField | Property |  | SynonymMapNames

### PR DESCRIPTION
**Pull request by heaths**
Fixes #12486

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 1]
```diff
@@ -212,9 +212,9 @@ public partial class AnalyzedTokenInfo
         public int StartOffset { get { throw null; } }
         public string Token { get { throw null; } }
     }
-    public partial class AnalyzeRequest
+    public partial class AnalyzeTextRequest
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:215]: Why not AnalyzeTextOptions?  I thought that's where we landed in the meeting.

Comment (heaths) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:215]: See my thread in our Teams channel. Basically, nothing about it is optional. According to [guidelines](https://azure.github.io/azure-sdk/general_design.html#client-interface):
> There are times when operations require the addition of optional data, provided in what is colloquially known as an “options bag”. Libraries should strive for consistent naming.
But `Text` is not optional, and one of `Analyzer` or `Tokenizer` is required by the service. The other option is, then, to have overloads requiring one or the other, but:
1. Not all languages can do that, and the type (string) would be the same anyway, and
2. That sounds like client-side validation of non-path arguments, which is against [guidelines](https://azure.github.io/azure-sdk/general_implementation.html#parameter-validation).
> DO NOT validate service parameters. This includes null checks, empty strings, and other common validating conditions. Let the service validate any request parameters.
If you or @KrzysztofCwalina have a different interpretation or a better suggestion, I'm happy to make a change.

Comment (heaths) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:215]: /cc @sima-zhu @alzimmermsft @xiangyan99 @sarangan12 @xirzec

Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:215]: Here's the C# specific guidance:  https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-parameters  (The example code has a bug where `BlobCreateOptions` doesn't have `string BlobName` but Krzysztof is going to fix that.)

We had a meeting earlier this week to clarify it a bit in relation to Java.  My current understanding is that `*Options` doesn't mean optional.  Here's what I'd expect now:

```C#
AnalyzeText(string text, Analyzer analyzer);
AnalyzeText(string text, Tokenizer tokenizer);
AnalyzeText(AnalyzeTextOptions options);

public class AnalyzeTextOptions
{
    public AnalyzeTextOptions(string text, Analyzer analyzer);
    public AnalyzeTextOptions(string text, Tokenizer tokenizer);
    public string Text { get; }
    public Analyzer Analyzer { get; }
    public Tokenizer Tokenizer { get; }
    public object WhateverElseMutable { get; set; }
}
```

I think it's too late to change this consistently everywhere, but I see no harm in calling the type `AnalyzeTextOptions`.  I don't feel strongly though so I'll defer to Krzysztof.

Comment (sima-zhu) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:215]: Java has made the change to AnalyzeTextRequest. 
@JonathanGiles could you suggest the best naming for Java?

Comment (sima-zhu) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:215]: [Guidance ](https://azure.github.io/azure-sdk/java_design.html#model-types)for java

Comment (heaths) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:215]: Java is a bit looser in their description:
> Optional parameters and settings to an operation should be collected into an options bag named <operation>Options

As such, I'm fine renaming this to `AnalyzeTextOptions` but not the overloads, which is too much like client-side validation of model properties. The swagger lists only `text` as required, though it probably should've used `oneOf` for the others (though, I'm not sure generators support that - just `required`). We can always add it later if we see a lot of confusion by customers.

Comment (sima-zhu) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:215]: https://github.com/Azure/azure-sdk-for-java/pull/11783

Comment (KrzysztofCwalina) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:215]: "Options" in the context of the option bag parameter names does not mean "optional". It's more like options on the fix prix menu, where you need to pick at least one option or you will end up hungry, yet with ah check to pay.

Comment (heaths) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:215]: @KrzysztofCwalina should we also have overloads (either method and/or model constructor overloads) that take either an `Analyzer` or `Tokenizer`? That really trends toward client-side validation of properties that aren't listed in swagger's `required`, but could be helpful. If so, I can get bugs opened across languages immediately.

Comment (heaths) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:215]: I'm going to send out an email to clarify our intent and open bugs across languages.

File: sdk/search/Azure.Search.Documents/CHANGELOG.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 - Moved models for managing indexes, indexers, and skillsets to `Azure.Search.Documents.Indexes.Models`.
 - Split `SearchServiceClient` into `SearchIndexClient` for managing indexes, and `SearchIndexerClient` for managing indexers, both of which are now in `Azure.Search.Documents.Indexes`.
-- Renamed `AnalyzeRequest` to `AnalyzeTextRequest`.
+- Renamed `AnalyzeRequest` to `AnalyzeTextOptions`, and overloaded constructors to make constructing it
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/CHANGELOG.md:9]: Nit - forgot the end of this

**Review by tg-msft — DISMISSED** [review]

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 3]
```diff
@@ -109,8 +109,8 @@ public partial class SearchIndexClient
...
         public virtual string ServiceName { get { throw null; } }
-        public virtual Azure.Response<System.Collections.Generic.IReadOnlyList<Azure.Search.Documents.Indexes.Models.AnalyzedTokenInfo>> AnalyzeText(string indexName, Azure.Search.Documents.Indexes.Models.AnalyzeRequest analyzeRequest, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.Response<System.Collections.Generic.IReadOnlyList<Azure.Search.Documents.Indexes.Models.AnalyzedTokenInfo>>> AnalyzeTextAsync(string indexName, Azure.Search.Documents.Indexes.Models.AnalyzeRequest analyzeRequest, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<System.Collections.Generic.IReadOnlyList<Azure.Search.Documents.Indexes.Models.AnalyzedTokenInfo>> AnalyzeText(string indexName, Azure.Search.Documents.Indexes.Models.AnalyzeTextOptions request, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:112]: Nit - these param names should probably be `options`

**Review by tg-msft — APPROVED** [review]

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs (hunk 3 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:112]: Huh. I could swear I changed those I wonder if that was another incarnation that got reverted. Thanks, though. I'll get that changed.

**Pull request merged**